README.md

```diff
@@ -66,6 +66,23 @@ install](https://pip.pypa.io/en/stable/reference/pip_install/#editable-installs)
 pip install -e ./instrumentation/opentelemetry-instrumentation-{integration}
 ```
 
+## Releasing
+
+Maintainers aim to periodically release new versions of the packages in `opentelemetry-python-contrib`.
+
+Contributions that enhance OTel for Python are welcome to be hosted upstream for the benefit of group collaboration. However, as a general rule, releases for packages in `opentelemetry-python-contrib` will be in a perpetual `~=0.X.0` (pre-1.0) version state. This is because maintainers and approvers cannot feasibly provide the stability guarantees that are implied with a 1.0 release for so many packages.
```

Comment:
I find this and the next paragraph a bit hard to understand, the way they are written kind of make me believe that generally, the contrib packages will be perpetually in `0.X.0`. This can be a bit misleading because we may perfectly end up having a codeowner for every package, thus having every package in a `A.B.C` version where `A > 0`.

An explanation on why we (maintainers and approvers) can't be codeowners for every contrib package (and thus many may remain as `0.X.Y`)  is fine, but I think a short comment and the list of requirements below is enough for our purposes.

Reply:
> the way they are written kind of make me believe that generally, the contrib packages will be perpetually in `0.X.0`

Yes that is the intention as we discussed in the last SIG meeting. Contrib packages will stay in `0.X.0` UNLESS someone becomes a CODEOWNER and wants to take it to `1.0`.

That's why we include the following paragraph right after:

```
To resolve this, members of the community are encouraged to commit to becoming a CODEOWNER for packages in `-contrib` that they feel experienced enough to maintain. CODEOWNERS can then follow the checklist below to release `-contrib` packages as 1.0 stable:
```

> This can be a bit misleading because we may perfectly end up having a codeowner for every package, thus having every package in a A.B.C version where A > 0.

I'm not sure if we discussed this. Even if we find CODEOWNER's for every package, the previous paragraph should still be taken (as it is written) "as a general rule" because when new packages get added, they won't have CODEOWNERs and so it will be in `0.X.0`.

> but I think a short comment and the list of requirements below is enough for our purposes.

Let me know if you have suggestions on how the paragraph can be more clear! 🙂 I tried to say it as concisely as possible but definitely open to feedback on how else it can be reworded!

Reply:
I am now even thinking that having a codeowner is a necessary but not sufficient condition for a package to step from 0.X.Y to 1.X.Y. Also, packages may get to a version greater than 0.X.Y and then they may lose their codeowner. I think both these points should be included in the explanation in this document.

Reply:
I agree that having a CODEOWNER is not a sufficient condition to release a package as 1.0. I think we are tracking that in #591 under "Remaning Issues". This PR just resolves #590 which is one of the issues. Once they are all resolved, we'll have a final path forward for releasing contrib packages as 1.0

> Also, packages may get to a version greater than 0.X.Y and then they may lose their codeowner. I think both these points should be included in the explanation in this document.

Are you referring to when CODEOWNERs leave the project? I didn't add anything about that in this file because we don't really have precedent regarding what we should do... It shouldn't be a major problem because once a package is release as 1.0 we won't have to "un-release" it. They can leave the project and the work can just stay at "1.X" or something and we won't update until we find a _new_ CODEOWNER.

I don't think we need to answer the question of them leaving just yet, but let me know what points you would like to see if you think otherwise?

Reply:
I'm not sure about document this as stated either. I don't think releasing 1.0 should depend on the package having a codeowner. We could totally review a package and release 1.0 for it if we feel confident about the public API.

Reply:
> I don't think releasing 1.0 should depend on the package having a codeowner. We could totally review a package and release 1.0 for it if we feel confident about the public API.

Hm, we had said in the [June 30 2020 SIG Meeting Google Docs](https://docs.google.com/document/d/1CIMGoIOZ-c3-igzbd6_Pnxx1SjAkjwqoYSUWxPY8XIs/edit#heading=h.wrlwshcxe9fm) that having a CODEOWNER was a requirement, because we don't want the maintainers to be responsible for so many packages that they can't hope to know everything about.

If we don't require a CODEOWNER then we don't really need this PR. But then are suggestion we removing this requirement from #591? Can anyone (not a CODEOWNER) ask a maintainer to release a package as 1.0 at that point because they need a stable version to be released?

Reply:
> Can anyone (not a CODEOWNER) ask a maintainer to release a package as 1.0 at that point because they need a stable version to be released?

I think this should be possible. I'm not sure how not releaing 1.0 for an "unmaintained" package will be a problem. If a package does not have a codeowner (dedicated maintainer) then it'll possibly suffer by not receiving fixes/enhancements in a timely manner irrespective of whether it has a stable release or not. I feel we are mixing up stability with support. We can release stable version meaning we guarantee we won't make breaking API changes to the package but it doesn't promise support for future bug fixes/enhancements IMO.

Reply:
> We can release stable version meaning we guarantee we won't make breaking API changes to the package but it doesn't promise support for future bug fixes/enhancements IMO.

What you said makes sense, we agreed in the SIG meetings that a CODEOWNER is not required, but is encouraged so this PR is updated to reflect that.

Reply:
> On the other hand, I'd be very reluctant to use a 0.X. release as it indicates constant churn.

Sorry do you mean to literally take a dependency as `opentelemetry-instrumentation-flask~=0.0`? (Implying I don't care if it is `0.24.0` or `0.25.0` for example). If that's what you meant then yeah I agree, I would be reluctant to take that kind of dependency. I was thinking they would take a more strict dependency like `~=0.24.0` if they know it works for them. Then they can ignore the future `0.X.0` updates if they want.

I can remove the "perpetual 0.x" statement, but I think that's more so describing what we _already do_ rather than suggesting a new pattern. Do you mean that in the near future we will stop releasing all the packages together, moving them to the next `0.X` even though they have no changes? That line was more so just to explain what we are doing right now, but if we won't do it anymore then we can definitely remove that comment.

Reply:
My only feedback is that we should not say that packages will _generally_ never hit 1.0 as that is not true. We may or may not release a 1.0 for any package in contrib depending on whether we feel confident about not making breaking changes in future. In another comment you mentioned this:

> With that said, maintainers will look for things like good documentation, good unit tests, and in general their own confidence when deciding to release a package as 1.0.

Which to me sounds something very different from the following

> However, as a general rule, releases for packages in `opentelemetry-python-contrib` will be in a perpetual `~=0.X.0` (pre-1.0) version state.

Reply:
I understand what you're saying, I've updated the PR to remove the "perpetual" `1.0` statement and used the quote you mentioned.